lib/ansible/plugins/connection/nspawn.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import os
+import pipes
+import subprocess
+import traceback
+import shlex
+
+from ansible import constants as C
+from ansible.errors import AnsibleError
+from ansible.plugins.connection import ConnectionBase
+from ansible.module_utils.basic import is_executable
+from ansible.utils.unicode import to_bytes
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+BUFSIZE = 65536
+
+
+class Connection(ConnectionBase):
+    ''' Local nspawn based connections '''
+
+    transport = 'nspawn'
+    has_pipelining = True
+    become_methods = frozenset(C.BECOME_METHODS)
+
+    def __init__(self, play_context, new_stdin, *args, **kwargs):
+        super(Connection, self).__init__(play_context, new_stdin,
+                *args, **kwargs)
+
+        display.vvv("NSPAWN ARGS %s" % self._play_context.nspawn_args)
+
+        self.chroot = self._play_context.remote_addr
+
+        if os.geteuid() != 0:
+            raise AnsibleError("nspawn connection requires running as root")
+
+        # we're running as root on the local system so do some
+        # trivial checks for ensuring 'host' is actually a chroot'able dir
+        if not os.path.isdir(self.chroot):
+            raise AnsibleError("%s is not a directory" % self.chroot)
+
+        chrootsh = os.path.join(self.chroot, 'bin/sh')
+        if not is_executable(chrootsh):
+            raise AnsibleError("%s does not look like a chrootable dir (/bin/sh missing)" % self.chroot)
```

Comment:
I'm unsure of systemd-nspawn terminology.  Is "chroot", "chrootable", etc, in descriptions the proper words to use?